inspector/src/components/graph/canvasGraphService.ts

```diff
@@ -256,6 +265,103 @@ export class CanvasGraphService {
 
         return startingPixel + normalizedValue * spaceAvailable;
     }
+
+    /**
+     * Add in any necessary event listeners.
+     * 
+     * @param canvas The canvas we want to attach listeners to.
+     */
+    private _attachEventListeners(canvas: HTMLCanvasElement) {
+        canvas.addEventListener("wheel", this._handleZoom);
+    }
+
+    /**
+     * We remove all event listeners we added.
+     * 
+     * @param canvas The canvas we want to remove listeners from.
+     */
+    private _removeEventListeners(canvas: HTMLCanvasElement) {
+        canvas.removeEventListener("wheel", this._handleZoom);
+    }
+
+    /**
+     * The handler for when we want to zoom in and out of the graph.
+     * 
+     * @param event a mouse wheel event.
+     */
+    private _handleZoom = (event: WheelEvent) => {
+        event.preventDefault();
+        
+        if (!event.deltaY) {
+            return;
+        }
+
+        const amount = (event.deltaY * -0.01 | 0) * 100;
+        const minZoom = 60;
+
+        // The max zoom is the largest dataset's length.      
+        const maxZoom = this.datasets.map((dataset: IPerfDataset) => dataset.data.length)
+                        .reduce((maxLengthSoFar: number, currLength: number) => {
+                            return Math.max(currLength, maxLengthSoFar)
+                        }, 0);
+
+        // Bind the zoom between [minZoom, maxZoom]
+        this._sizeOfWindow = Math.min(Math.max(this._sizeOfWindow - amount, minZoom), maxZoom);
+    }
+
+    /**
+     * Will generate a playhead with a futurebox that takes up (1-scalefactor)*100% of the canvas.
+     * 
+     * @param drawableArea The remaining drawable area.
+     * @param scaleFactor The Percentage between 0.0 and 1.0 of the canvas the data gets drawn on.  
+     */
+    private _drawPlayheadRegion(drawableArea: IGraphDrawableArea, scaleFactor: number) {
+        const { _ctx: ctx } = this;
+       
+        if (!ctx) {
+            return;
+        }
+
+        const dividerSize = 2;

```

Comment:
would be great to extract constants at the top

Reply:
Done